src/sync/update_tx.js

```diff
@@ -1,157 +1,149 @@
-const { Qweb3 } = require('qweb3');
-const fetch = require('node-fetch');
 
-const config = require('../config/config');
 const logger = require('../utils/logger');
+const blockchain = require('../api/blockchain');
+const centralizedOracle = require('../api/centralized_oracle');
+const decentralizedOracle = require('../api/decentralized_oracle');
+const DBHelper = require('../db/db_helper');
 
-const qclient = new Qweb3(config.QTUM_RPC_ADDRESS);
-
-async function checkTxStatus(txid) {
-  const resp = await qclient.getTransactionReceipt(txid);
-  console.log(resp);
-  if (_.isEmpty(resp)) {
-    return {
-      status: 'PENDING',
-    };
-  }
-
-  const tx = resp[0];
-  if (_.isEmpty(tx.log)) {
-    return {
-      status: 'FAIL',
-      gasUsed: tx.gasUsed,
-      blockNum: tx.blockNum,
-    };
-  }
-
-  return {
-    status: 'SUCCESS',
-    gasUsed: tx.gasUsed,
-    blockNum: tx.blockNum,
-  };
-}
-
-async function updateTxDB(db) {
+async function updatePendingTxs(db) {
   let pendingTxs;
   try {
     pendingTxs = await db.Transactions.cfind({ status: 'PENDING' })
       .sort({ createTime: -1 }).exec();
   } catch (err) {
-    logger.error(`Error updateTxDB: ${err.message}`);
+    logger.error(`Error: get pending Transactions: ${err.message}`);
     throw err;
   }
 
   // TODO(frank): batch to void too many rpc calls
-  await Promise.all(pendingTxs.map(async (txid) => {
-    await updateTx(txid, db);
-  }));
+  const updatePromises = [];
+  _.each(pendingTxs, (tx) => {
+    updatePromises.push(new Promise(async (resolve) => {
+      await updateTx(tx);
+      await updateDB(tx, db);
+      await executeFollowUpTx(tx, db);
+      resolve();
+    }));
+  })
+  await Promise.all(updatePromises);
 }
 
-async function updateTx(approveTxid, db) {
-  const txReceipt = await checkTxStatus(approveTxid);
-  let tx;
-  if (txReceipt.status !== 'PENDING') {
+// Update the Transaction info
+async function updateTx(tx) {
+  const resp = await blockchain.getTransactionReceipt({ transactionId: tx._id });
+
+  if (_.isEmpty(resp)) {
+    tx.status = 'PENDING';
+  } else if (_.isEmpty(resp[0].log)) {
+    tx.status = 'FAIL';
+    tx.gasUsed = resp[0].gasUsed;
+    tx.blockNum = resp[0].blockNumber;
+  } else {
+    tx.status = 'SUCCESS';
+    tx.gasUsed = resp[0].gasUsed;
+    tx.blockNum = resp[0].blockNumber;
+  }
+}
+
+// Update the DB with new Transaction info
+async function updateDB(tx, db) {
+  if (tx.status !== 'PENDING') {
     try {
-      tx = await db.Transactions.update(
-        { address: approveTxid },
+      logger.debug(`Update: Transaction ${tx.type} txid:${tx._id}`);
+      await db.Transactions.update(
+        { _id: tx._id },
         {
           $set: {
-            status: txReceipt.status,
-            gasUsed: txReceipt.gasUsed,
-            blockNum: txReceipt.blockNum,
+            status: tx.status,
+            gasUsed: tx.gasUsed,
+            blockNum: tx.blockNum,
           },
-        }, {},
+        }, 
+        {},
```

Comment:
`returnUpdatedDocs: True` according to https://github.com/louischatriot/nedb Updating documents section

Reply:
👍